src/isolated-runner/RetryDecorator.ts

```diff
@@ -0,0 +1,60 @@
+import { EventEmitter } from 'events';
+import { TestRunner, RunOptions, RunResult, RunStatus } from 'stryker-api/test_runner';
+import { isErrnoException, errorToString } from '../utils/objectUtils';
+import TestRunnerDecorator from './TestRunnerDecorator';
+import Task from '../utils/Task';
+
+const BROKEN_PIPE_ERROR_CODE = 'EPIPE';
+const ERROR_MESSAGE = 'Test runner crashed. Tried twice to restart it without any luck. Last time the error message was: ';
+
+/**
+ * Wraps a test runner and implements the retry functionality.
+ */
+export default class RetryDecorator extends TestRunnerDecorator {
+  private currentRunTask: Task<RunResult>;
+
+  run(options: RunOptions): Promise<RunResult> {
```

Comment:
I'm not sure i follow your logic. The `RunOptions` here are provided in the `run` method as specified on the `TestRunner` interface.